proposals/NNNN-resources-get-dimensions.md

```diff
@@ -0,0 +1,217 @@
+---
+title: "[NNNN] - GetDimensions mapping to builtins and intrinsics"
+params:
+  authors:
+    - github_username: hekota
+  status: Under Consideration
+  sponsors:
+    - github_username: hekota
+---
+
+## Introduction
+
+All buffer and texture resources have a `GetDimensions` member function and,
+which must be supported across all resource types. There are many different
+overloads of this function. This proposal summarizes the variants and outlines
+how they can be implemented in Clang using built-in functions, as well as how
+they map to LLVM intrinsics.
+
+## Motivation
+
+`GetDimensions` member function on resource classes is part of the HLSL language
+and we need to support it in Clang.
+
+## Proposed solution
+
+### Clang builtin functions
+
+There are 54 `GetDimensions` member function overloads across all resource
+classes. We need to add a number of builtin functions to Clang that will be used
+to implement these. It will most likely amount to one builtin function per
+unique argument list, altough the same builtin function could be used for
+`GetDimensions` overloads that different only in `uint` vs. `float` argument
+types that are otherwise doing the same thing. This should greatly reduce the
+number of needed builtins function.
```

Comment:
I rewrote this paragraph to read better in my head. Feel free to use it!
```suggestion
There are 54 `GetDimensions` member function overloads across all resource
classes. We need to add a number of built-in functions to Clang to implement 
these. This will most likely amount to one built-in function per unique argument
list. However, `GetDimensions` overloads that differ only by the types of their
arguments (namely, `uint` vs. `float`) can be consolidated under the same built-in
function, thereby greatly reducing the number of needed built-in functions.
```

Otherwise, you just misspelt the word "altough".

Also note: you used "built-in" on line 16 but "builtin" everywhere else. You could find and replace "builtin" with "built-in" to be more consistent.